src/Microsoft.IdentityModel.Tokens/Microsoft.IdentityModel.Tokens.csproj

```
@@ -57,7 +57,7 @@
     <PackageReference Include="System.Diagnostics.DiagnosticSource" Version="$(SystemDiagnosticSourceVersion)" />
   </ItemGroup>
 
-  <ItemGroup Condition="'$(TargetFrameworkIdentifier)' != '.NETCoreApp'">
+  <ItemGroup Condition="'$(TargetFrameworkIdentifier)' != '.NETCoreApp' And !$(TargetFramework.StartsWith('net9.0'))">
```

Our NetCoreApp target fw's are [net6, net8, and net9](https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/blob/35caa6cb0b693188b5c5c1b9434414f1a3ba43ba/build/targets.props#L3C83-L3C104). Consider being explicit like:  
```suggestion
  <ItemGroup Condition="$(TargetFramework) == 'net6.0' or $(TargetFramework) == 'net8.0'">
```

---

thank you! I've updated!

---

Wait, Kelly, I tried changing the condition to   `<ItemGroup Condition="'$(TargetFramework)'== 'net6.0' or '$(TargetFramework)' == 'net8.0'">: `

Then I got a bunch of build errors for the three `csproj` files updated with the condition above. The build errors were from all TFMs and were of two categories:

1. cannot find dlls
2. `The type or namespace name 'JsonIgnore' could not be found (are you missing a using directive or an assembly reference?`

I tried researching why and found this thru copilot hah: 
`The error occurs because in .NET 9.0, even though System.Text.Json is included as part of the framework, you need to add an explicit reference to the assembly through a using directive. This is due to how JsonIgnore and other Json-related attributes are organized in .NET.`

The explicit reference I added to the csproj files was:
```
<ItemGroup Condition="'$(TargetFramework)' == 'net9.0'">
  <FrameworkReference Include="Microsoft.AspNetCore.App" />
</ItemGroup>
```

This change still caused build errors similar to above, so now I updated the reference as follows: 
```
  <ItemGroup Condition="'$(TargetFrameworkIdentifier)' != '.NETCoreApp' And '$(TargetFramework)' != 'net9.0' ">
    <PackageReference Include="System.Text.Json" Version="$(SystemTextJson)" />
  </ItemGroup>
```

What do you think?

---

Hi Mallika! I pulled your branch and tried things out. Sorry, I misread the initial line and interpreted it as `$(TargetFramework) == NetCoreApp` instead of `!=`. 

You can remove the second half of the statement because any fw that's != NetCoreApp won't start with net9.0 and I've verified it locally

---

All good! thank you! I removed it